File: 01.DefiningClasses/DefiningClassHomework/DefiningClass/LaptopShop/Laptop.cs
using System;

class Laptop
{
    private string model;
    private string manufacturer;
    private string processor;
    private string ram;
    private string graphicsCard;
    private string hdd;
    private string screen;
    private double price;

    public string Model
    {
        get { return this.model; }
        set
        {
            if (value != null && value.Trim() != String.Empty)
            {
                this.model = value;
            }
            else
            {
                throw new ArgumentException("This is not valid model for laptop!");
            }
        }
    }
    public double Price
    {
        get { return this.price; }
        set
        {
            if (value > 0)
            {
                this.price = value;
            }
            else
            {
                throw new Exception("This is not valid value for double numbers!");
            }
        }
    }
    public string Manufacturer {
        get { return this.manufacturer; }
        set
        {
            if (value != null)
            {
                this.manufacturer = value;
            }
        }
    }
    public string Processor
    {
        get { return this.processor; }
        set
        {
            if (value != null)
            {
                this.processor = value;
            }
        }
    }
    public string Ram
    {
        get { return this.ram; }
        set
        {
            if (value != null)
            {
                this.ram = value;
            }
        }
    }
    public string GraphicsCard
    {
        get { return this.graphicsCard; }
        set
        {
            if (value != null)
            {
                this.graphicsCard = value;
            }
        }
    }
    public string HDD
    {
        get { return this.hdd; }
        set
        {
            if (value != null)
            {
                this.hdd = value;
            }
        }
    }
    public string Screen
    {
        get { return this.screen; }
        set {
            if (value != null)
            {
                this.screen = value;
            }
        }
    }
    
    public Laptop(string model, double price)
    {
        this.Model = model;
        this.Price = price;
        PrintInfo();
    }

    public Laptop(  string model, double price, string manufacturer = null
                    , string processor = null, string ram = null, string graphicsCard = null
                    , string hdd = null, string screen = null, string battery = null
                    , string baterryLife = null)
    {
        this.Model = model;
        this.Price = price;
        this.Manufacturer = manufacturer;
        this.Processor = processor;
        this.Ram = ram;
        this.GraphicsCard = graphicsCard;
        this.HDD = hdd;
        this.Screen = screen;
        BatteryInfo batteryInfo = new BatteryInfo(battery, baterryLife);
        PrintInfo("full");
    }

    private void PrintInfo(string information)
    {
        Console.WriteLine("model {0}", this.model);
        Console.WriteLine("manufacturer {0}", this.manufacturer);
        Console.WriteLine("processor {0}", this.processor);
        Console.WriteLine("RAM {0}", this.ram);
        Console.WriteLine("graphics card {0}", this.graphicsCard);
        Console.WriteLine("HDD {0}", this.hdd);
        Console.WriteLine("screen {0}", this.screen);
        Console.WriteLine("battery {0}");
        Console.WriteLine("battery life {0}");
        Console.WriteLine("price {0:f2}lv", this.price);
    }

    private void PrintInfo()
    {
        Console.WriteLine("model {0}", this.model);
        Console.WriteLine("price {0:f2}lv", this.price);
    }
}

File: 01.DefiningClasses/OtherHomeworks/DefClass-Java/src/com/homework/store/Laptop.java
package com.homework.store;

public class Laptop {

	private String model;
	private String manufacturer;
	private String processor;
	private String ram;
	private String graphicsCard;
	private String hdd;
	private String screen;
	private Battery battery;
	private float batteryLifeHours;
	private double price;

	public Laptop(String model, double price) throws Exception {
		setModel(model);
		setPrice(price);
	}

	public Laptop(String model, String manufacturer, String processor,
			String ram, String graphicsCard, String hDD, String screen,
			Battery battery, float batteryLifeHours, double price)
			throws Exception {
		this(model, price);
		setManufacturer(manufacturer);
		setProcessor(processor);
		setRam(ram);
		setGraphicsCard(graphicsCard);
		setHDD(hDD);
		setScreen(screen);
		setBattery(battery);
		setBatteryLifeHours(batteryLifeHours);
	}

	public String getModel() {
		return model;
	}

	public void setModel(String model) throws Exception {
		if (model == null || model.trim().length() == 0) {
			throw new Exception();
		}
		this.model = model;
	}

	public String getManufacturer() {
		return manufacturer;
	}

	public void setManufacturer(String manufacturer) throws Exception {
		if (manufacturer == null || manufacturer.trim().length() == 0) {
			throw new Exception();
		}
		this.manufacturer = manufacturer;
	}

	public String getProcessor() {
		return processor;
	}

	public void setProcessor(String processor) throws Exception {
		if (processor == null || processor.trim().length() == 0) {
			throw new Exception();
		}
		this.processor = processor;
	}

	public String getRam() {
		return ram;
	}

	public void setRam(String ram) throws Exception {
		if (ram == null || ram.trim().length() == 0) {
			throw new Exception();
		}
		this.ram = ram;
	}

	public String getGraphicsCard() {
		return graphicsCard;
	}

	public void setGraphicsCard(String graphicsCard) throws Exception {
		if (graphicsCard == null || graphicsCard.trim().length() == 0) {
			throw new Exception();
		}
		this.graphicsCard = graphicsCard;
	}

	public String getHDD() {
		return hdd;
	}

	public void setHDD(String hDD) throws Exception {
		if (hDD == null || hDD.trim().length() == 0) {
			throw new Exception();
		}
		hdd = hDD;
	}

	public String getScreen() {
		return screen;
	}

	public void setScreen(String screen) throws Exception {
		if (screen == null || screen.trim().length() == 0) {
			throw new Exception();
		}
		this.screen = screen;
	}

	public Battery getBattery() {
		return battery;
	}

	public void setBattery(Battery battery) throws Exception {
		if (battery == null) {
			throw new Exception();
		}

		this.battery = battery;
	}

	public float getBatteryLifeHours() {
		return batteryLifeHours;
	}

	public void setBatteryLifeHours(float batteryLifeHours) throws Exception {
		if (batteryLifeHours < 0f) {
			throw new Exception();
		}

		this.batteryLifeHours = batteryLifeHours;
	}

	public double getPrice() {
		return price;
	}

	public void setPrice(double price) throws Exception {
		if (batteryLifeHours < 0) {
			throw new Exception();
		}

		this.price = price;
	}

	public String toString() {
		return "Laptop [model=" + model + ", manufacturer=" + manufacturer
				+ ", processor=" + processor + ", ram=" + ram
				+ ", graphicsCard=" + graphicsCard + ", hdd=" + hdd
				+ ", screen=" + screen + ", battery=" + battery.toString()
				+ ", batteryLifeHours=" + batteryLifeHours + ", price=" + price
				+ "]";
	}

}
